src/c++/perf_analyzer/client_backend/triton_c_api/triton_loader.cc

```diff
@@ -921,9 +926,20 @@ TritonLoader::Infer(
   timer.CaptureTimestamp(tc::RequestTimers::Kind::SEND_END);
   // Wait for the inference to complete.
   TRITONSERVER_InferenceResponse* completed_response = completed.get();
-  RETURN_IF_TRITONSERVER_ERROR(
-      GetSingleton()->inference_response_error_fn_(completed_response),
-      "response status");
+
+  // check if there completed response is an error and needs to shut down
+  // gracefully
+  TRITONSERVER_Error* completed_response_err =
+      GetSingleton()->inference_response_error_fn_(completed_response);
+  if (completed_response_err != nullptr) {
+    // intentionally not using the return value from Clean up here
+    // it is captured to avoid warnings but at this point, the error from the
+    // tritonserver (completed_response_err) is more important to bubble up
+    Error val = CleanUp(completed_response, allocator);
```

Comment:
val doesn't seem to be used.

Reply:
that was intentional and is commented in the line above